### PR TITLE
Filtering notebook

### PR DESCRIPTION
**Pull request by wasade**
The filtering notebook for removing bloom sequences. Rendered version of the notebook is [here](http://nbviewer.ipython.org/github/wasade/American-Gut/blob/e9f79334c41da6f5184e9372473eca076398507d/ipynb/FilterAndPickOTUs.ipynb) and the updated index is [here](http://nbviewer.ipython.org/github/wasade/American-Gut/blob/e9f79334c41da6f5184e9372473eca076398507d/ipynb/index.ipynb).

### CODE REVIEW

File: ipynb/FilterAndPickOTUs.ipynb [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+     "source": [
+      "* License: BSD\n",
+      "* Copyright: American Gut Project, 2014"
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Early in the American Gut Project, it was observed that some organisms bloomed likely as a result of increased shipping time and delay between when samples were collected and when they were put on ice (more detail can be found [here](http://americangut.org/?page_id=277)). The purpose of this notebook is to apply the filter developed in order to bioinformatically subtract these observed bloom sequences from fecal samples. It is important to apply this filter when combining data with the American Gut as to remove a potential study-effect bias as all fecal data in the American Gut has had this filter applied. The specific steps covered are:\n",
```
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: It's fairly normal, for instance, one of the typical uses is "bioinformatically removing human reads from WGS data"

File: ipynb/FilterAndPickOTUs.ipynb [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+      "cmd_prefix = 'source ~/.bash_profile; workon ag'\n",
+      "\n",
+      "# bootstrap the submit method\n",
+      "submit_8gb = partial(submit, cmd_prefix, prj_name, 'short8gb', '-l walltime=23:59:59')\n",
+      "submit_smr = partial(submit, cmd_prefix, prj_name, 'long8gb', '-l nodes=1:ppn=32 -l walltime=72:00:00')\n",
+      "\n",
+      "# setup a working directory\n",
+      "if not os.path.exists(working_dir):\n",
+      "    os.mkdir(working_dir)\n",
+      "path_joiner = lambda x: os.path.join(working_dir, x)"
```
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: Done

File: ipynb/FilterAndPickOTUs.ipynb [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+     "source": [
+      "Here is where we'll setup paths for our input datasets"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# setup the paths to your input datasets\n",
+      "paths_to_input_datasets = [\"/Users/mcdonadt/scratch/ICU/r1/seqs_r1.fna\",\n",
```
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: These are for the user to change, and their data may not be under the basedir (as is the case here)

File: ipynb/FilterAndPickOTUs.ipynb [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# path to the bloom sequences (note, this can be obtained from https://github.com/biocore/American-Gut/tree/master/data/AG\n",
+      "bloom_seqs = \"/Users/mcdonadt/ag/American-Gut/data/AG/BLOOM.fasta\"\n",
+      "\n",
+      "# path to Greengenes\n",
+      "reference_rep_set = '/scratch/Users/mcdonadt/greengenes/gg_13_8_otus/rep_set/97_otus.fasta' # e.g., path to 97_otus.fasta from Greengenes\n",
```
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: This is how it was done in qiime 1.8 and it is more general specifying the files individually, unless I'm misunderstanding something?

File: ipynb/FilterAndPickOTUs.ipynb [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+     "source": [
+      "On this system, all of the nodes have 64 cores, so let's allow SortMeRNA to use all of them."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# setup SortMeRNA params file for OTU picking\n",
+      "num_threads_otu_picking = 64\n",
```
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: Fixed

File: ipynb/FilterAndPickOTUs.ipynb [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+      "\n",
+      "for_otu_picking = []\n",
+      "for bfs in bloom_filtered_seqs:\n",
+      "    trimmed_file = bfs + '-%dnt' % trim_length\n",
+      "    for_otu_picking.append(bfs)\n",
+      "    for_otu_picking.append(trimmed_file)\n",
+      "    \n",
+      "    with open(trimmed_file, 'w') as trimmed, open(bfs) as to_trim:\n",
+      "        for payload in trimmer(to_trim):\n",
+      "            trimmed.write(\">%s\\n%s\\n\" % payload)"
```
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: Unsure, is this not clear though?

**Comment by wasade** [comment]
@josenavas, took a bit, but comments addressed. Good to go?

**Comment by josenavas** [comment]
Yup :+1: Do we need to devs here too?

**Comment by wasade** [comment]
It would be nice...

**Comment by josenavas** [comment]
@adamrp @ElDeveloper @squirrelo are any of you able to review this one?

File: ipynb/FilterAndPickOTUs.ipynb [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "%run cluster_utils.ipy\n",
+      "import os\n",
+      "from functools import partial\n",
+      "\n",
+      "current_home = os.environ['HOME']\n",
+      "basedir = os.path.join(current_home, \"ICU\")\n",
```
Comment (adamrp) [inline ipynb/FilterAndPickOTUs.ipynb]: Setting base dir as ICU intentional?

File: ipynb/FilterAndPickOTUs.ipynb [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+      "\n",
+      "# define a submit function against qsub\n",
+      "def submit(cmd_prefix, prj_name, queue, extra_args, cmd):\n",
+      "    \"\"\"Submission wrapper to submit to qsub\"\"\"\n",
+      "    cmd = \"%s; %s\" % (cmd_prefix, cmd)\n",
+      "    return submit_qsub(cmd, job_name=prj_name, queue=queue,\n",
+      "                       extra_args=extra_args)\n",
+      "\n",
+      "# define any environment specific details for any running compute job\n",
+      "cmd_prefix = 'source ~/.bash_profile; workon ag'\n",
```
Comment (adamrp) [inline ipynb/FilterAndPickOTUs.ipynb]: Assumes virtualenv active with an `ag` environment defined

**Comment by adamrp** [comment]
A couple minor comments, neither blocking. Do you want to comment/address them before I merge?

**Comment by wasade** [comment]
will make some modifications quickly, thanks!

File: ipynb/FilterAndPickOTUs.ipynb (hunk 7 above)
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: will add comment about this being likely necessary to change

File: ipynb/FilterAndPickOTUs.ipynb (hunk 8 above)
Comment (wasade) [inline ipynb/FilterAndPickOTUs.ipynb]: comment indicates it is to define any env specific details... would you like that expanded further?

**Comment by wasade** [comment]
ah, don't merge yet

**Comment by wasade** [comment]
good to go

**Comment by adamrp** [comment]
Thanks @wasade!

**Comment by wasade** [comment]
boom!

**Pull request merged**